src/ashell/jerry-code.c

```diff
@@ -195,53 +195,122 @@ static void eval_list(requires_list_t **list)
     }
 }
 
+static void skip_whitespace(char **ptr)
+{
+    // modifies: *ptr
+    //  effects: increments *ptr until it's past any whitespace (space, tab,
+    //             newline) or gets to null terminator
+    while (**ptr != '\0') {
+        if (**ptr == ' ' || **ptr == '\t' || **ptr == '\r' || **ptr == '\n') {
+            ++(*ptr);
+        }
+        else {
+            return;
+        }
+    }
+}
+
+static bool skip_char_with_whitespace(char **ptr, char match)
+{
+    // modifies: *ptr
+    //  effects: skips any whitespace in *ptr before and after a match char or
+    //             until null terminator is reached
+    //           if first non-whitespace char found is match, returns true;
+    //             otherwise, false
+    skip_whitespace(ptr);
+    if (**ptr != match) {
+        return false;
+    }
+    skip_whitespace(ptr);
```

Comment:
You need a ++(*ptr); before this line.  Otherwise skip_whitespace won't do anything as ptr is currently pointing at '('